setup.py

```diff
@@ -242,6 +242,29 @@ def build_extension(self, ext):
 
 
 def generate_cython():
+    # Check Cython version
+    from numpy.compat import _pep440
+    try:
+        # try the cython in the installed python first (somewhat related to scipy/scipy#2397)
+        import Cython
+        from Cython.Compiler.Version import version as cython_version
+    except ImportError as e:
+        # The `cython` command need not point to the version installed in the
+        # Python running this script, so raise an error to avoid the chance of
+        # using the wrong version of Cython.
+        msg = 'Cython needs to be installed in Python as a module'
+        raise OSError(msg) from e
+    else:
+        # Note: keep in sync with that in pyproject.toml
+        # Update for Python 3.10
+        required_version = '0.29.24'
+
+        if _pep440.parse(cython_version) < _pep440.Version(required_version):
+            cython_path = Cython.__file__
+            raise RuntimeError(f'Building {VENDOR} requires Cython >= {required_version}'
+                               f', found {cython_version} at {cython_path}')
+
+    # Process files
```

Comment:
The cython version check is moved here because importing _pep440 fails for package specific cythonizing when kept in the `tools/cythonize.py` module.